src/ir/value.h

```diff
@@ -114,6 +114,15 @@ class Value {
         value_);
   }
 
+  // A downcast operation. Just delegates directly to std::get on variants.
+  template<class T>
+  const T &As() const { return std::get<T>(value_); }
+
+  // A dynamic downcast operation. Just delegates directly to std::get_if on
+  // variants.
+  template<class T>
+  const T *If() const { return std::get_if<T>(&value_); }
+
```

Comment:
Are these needed anymore?

Reply:
I'm using `As` in some tests, either in this PR or follow-on PRs. I don't think I'm using `If` at the moment, but it seems like a useful thing to have if we have `As`. While visitors are indeed a better pattern for full visitation of a tree, sometimes it is nice to just get the single-step downcast to the specific type of value.

Reply:
A visitor would be ideal, but to unblock let us use `If<T>` variant. The `As<T>` variant can throw an exception and it is not obvious at use sites. With `If` variant, you will get a `nullptr`, and we can easily document use `CHECK_NOTNULL` to document expectations.

Also, add tests for `If` (in a follow up or separate PR?)

Reply:
OK, that sounds good.

The issue with a `Visitor` here is that these are `Value`s; as such, the `Value`s that are the predecessors to this `Value` are not a tree, but a graph - the dataflow graph. Creating a `Visitor` on the `Value`s would be akin to performing a dataflow graph traversal, and I'm not sure it makes sense to do so when we want to lean on Souffle for that.

Reply:
Using `If` instead of `As` in the test now.